### ISSUE TITLE
Error Msg During Installation of .NET Core SDK on Ubuntu 18.04.2: mysql-server-5.7 not configured

### ISSUE THREAD

**Issue by waqargithub** [dotnet/core#3672]
Problem encountered on https://dotnet.microsoft.com/learn/dotnet/hello-world-tutorial/install
Operating System: Linux Ubuntu 18.04 - x64

Provide details about the problem you are experiencing. Include your operating system version, exact error message, code sample, and anything else that is relevant.

--------------------------------

OS Version: Ubuntu 18.04.2

I opened a C# project for the first time in Visual Studio Code on this machine. I was prompted to install a C# plug-in.

I also got an error message saying .NET Core SDK was missing. Messages on Visual Studio Code:

```
[fail]: OmniSharp.MSBuild.ProjectLoader
        The reference assemblies for .NETFramework,Version=v4.5.2 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks
```

And I got another message in VS Code prompting me to install the .NET Core SDK. I clicked on the button to install it and was taken to a web page to Download .NET Core SDK for Visual Studio Code. I clicked on Install for Linux button and selected Ubuntu 18.04 x64 as my distribution. I then followed the steps (I clicked on the *copy* link at the top right of list of commands to be typed into the terminal and pasted into my terminal).

The content of the terminal window is pasted below.

```
ubuntu@ubuntu:~$ wget -q https://packages.microsoft.com/config/ubuntu/18.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
ubuntu@ubuntu:~$ sudo dpkg -i packages-microsoft-prod.deb
Selecting previously unselected package packages-microsoft-prod.
(Reading database ... 149336 files and directories currently installed.)
Preparing to unpack packages-microsoft-prod.deb ...
Unpacking packages-microsoft-prod (1.0-ubuntu18.04.2) ...
Setting up packages-microsoft-prod (1.0-ubuntu18.04.2) ...
ubuntu@ubuntu:~$ sudo add-apt-repository universe
'universe' distribution component enabled for all sources.
Ign:1 cdrom://Ubuntu 18.04.2 LTS _Bionic Beaver_ - Release amd64 (20190210) bionic InRelease
Hit:2 cdrom://Ubuntu 18.04.2 LTS _Bionic Beaver_ - Release amd64 (20190210) bionic Release
Ign:4 http://dl.google.com/linux/chrome/deb stable InRelease                   
Hit:5 http://dl.google.com/linux/chrome/deb stable Release                     
Hit:6 http://security.ubuntu.com/ubuntu bionic-security InRelease              
Hit:7 https://packages.microsoft.com/repos/vscode stable InRelease             
Get:9 https://packages.microsoft.com/ubuntu/18.04/prod bionic InRelease [3982 B]
Hit:10 http://archive.ubuntu.com/ubuntu bionic InRelease                       
Get:11 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [613 kB]
Get:12 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]     
Get:13 https://packages.microsoft.com/ubuntu/18.04/prod bionic/main amd64 Packages [75.4 kB]
Hit:14 http://ppa.launchpad.net/webupd8team/java/ubuntu bionic InRelease       
Hit:15 https://dl.yarnpkg.com/debian stable InRelease                          
Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages [8570 kB]
Get:17 http://security.ubuntu.com/ubuntu bionic-security/universe Translation-en [204 kB]
Get:18 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 DEP-11 Metadata [42.0 kB]
Get:19 http://security.ubuntu.com/ubuntu bionic-security/universe DEP-11 48x48 Icons [16.4 kB]
Get:20 http://security.ubuntu.com/ubuntu bionic-security/universe DEP-11 64x64 Icons [111 kB]
Get:21 http://archive.ubuntu.com/ubuntu bionic/universe Translation-en [4941 kB]
Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 DEP-11 Metadata [3287 kB]
Get:23 http://archive.ubuntu.com/ubuntu bionic/universe DEP-11 48x48 Icons [2151 kB]
Get:24 http://archive.ubuntu.com/ubuntu bionic/universe DEP-11 64x64 Icons [8420 kB]
Get:25 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 DEP-11 Metadata [295 kB]
Get:26 http://archive.ubuntu.com/ubuntu bionic-updates/main DEP-11 48x48 Icons [73.8 kB]
Get:27 http://archive.ubuntu.com/ubuntu bionic-updates/main DEP-11 64x64 Icons [143 kB]
Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [1012 kB]
Get:29 http://archive.ubuntu.com/ubuntu bionic-updates/universe Translation-en [312 kB]
Get:30 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 DEP-11 Metadata [254 kB]
Get:31 http://archive.ubuntu.com/ubuntu bionic-updates/universe DEP-11 48x48 Icons [205 kB]
Get:32 http://archive.ubuntu.com/ubuntu bionic-updates/universe DEP-11 64x64 Icons [456 kB]
Fetched 31.3 MB in 20s (1599 kB/s)                                             
Reading package lists... Done
W: Target Packages (main/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Packages (main/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Translations (main/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11 (main/dep11/Components-amd64.yml) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11 (main/dep11/Components-all.yml) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11-icons-small (main/dep11/icons-48x48.tar) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11-icons (main/dep11/icons-64x64.tar) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target CNF (main/cnf/Commands-amd64) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target CNF (main/cnf/Commands-all) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Packages (main/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Packages (main/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Translations (main/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11 (main/dep11/Components-amd64.yml) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11 (main/dep11/Components-all.yml) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11-icons-small (main/dep11/icons-48x48.tar) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11-icons (main/dep11/icons-64x64.tar) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target CNF (main/cnf/Commands-amd64) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target CNF (main/cnf/Commands-all) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
ubuntu@ubuntu:~$ sudo apt-get update
Ign:1 cdrom://Ubuntu 18.04.2 LTS _Bionic Beaver_ - Release amd64 (20190210) bionic InRelease
Hit:2 cdrom://Ubuntu 18.04.2 LTS _Bionic Beaver_ - Release amd64 (20190210) bionic Release
Ign:3 http://dl.google.com/linux/chrome/deb stable InRelease                   
Hit:4 http://dl.google.com/linux/chrome/deb stable Release                     
Hit:6 https://dl.yarnpkg.com/debian stable InRelease                           
Get:8 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]    
Hit:9 http://archive.ubuntu.com/ubuntu bionic InRelease                        
Hit:10 http://ppa.launchpad.net/webupd8team/java/ubuntu bionic InRelease       
Hit:11 https://packages.microsoft.com/repos/vscode stable InRelease            
Hit:12 https://packages.microsoft.com/ubuntu/18.04/prod bionic InRelease       
Hit:13 http://archive.ubuntu.com/ubuntu bionic-updates InRelease               
Get:14 http://security.ubuntu.com/ubuntu bionic-security/main amd64 DEP-11 Metadata [38.5 kB]
Get:15 http://security.ubuntu.com/ubuntu bionic-security/main DEP-11 48x48 Icons [17.6 kB]
Get:16 http://security.ubuntu.com/ubuntu bionic-security/main DEP-11 64x64 Icons [41.5 kB]
Get:17 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 DEP-11 Metadata [42.2 kB]
Get:18 http://security.ubuntu.com/ubuntu bionic-security/universe DEP-11 48x48 Icons [16.4 kB]
Get:19 http://security.ubuntu.com/ubuntu bionic-security/universe DEP-11 64x64 Icons [108 kB]
Fetched 353 kB in 4s (88.8 kB/s)                                
Reading package lists... Done
W: Target Packages (main/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Packages (main/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Translations (main/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11 (main/dep11/Components-amd64.yml) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11 (main/dep11/Components-all.yml) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11-icons-small (main/dep11/icons-48x48.tar) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11-icons (main/dep11/icons-64x64.tar) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target CNF (main/cnf/Commands-amd64) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target CNF (main/cnf/Commands-all) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Packages (main/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Packages (main/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target Translations (main/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11 (main/dep11/Components-amd64.yml) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11 (main/dep11/Components-all.yml) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11-icons-small (main/dep11/icons-48x48.tar) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target DEP-11-icons (main/dep11/icons-64x64.tar) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target CNF (main/cnf/Commands-amd64) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
W: Target CNF (main/cnf/Commands-all) is configured multiple times in /etc/apt/sources.list:6 and /etc/apt/sources.list.d/vscode.list:3
ubuntu@ubuntu:~$ sudo apt-get install apt-transport-https
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  apt-transport-https
0 upgraded, 1 newly installed, 0 to remove and 563 not upgraded.
2 not fully installed or removed.
Need to get 1692 B of archives.
After this operation, 153 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 apt-transport-https all 1.6.12 [1692 B]
Fetched 1692 B in 0s (4128 B/s)               
Selecting previously unselected package apt-transport-https.
(Reading database ... 149344 files and directories currently installed.)
Preparing to unpack .../apt-transport-https_1.6.12_all.deb ...
Unpacking apt-transport-https (1.6.12) ...
Setting up apt-transport-https (1.6.12) ...
Setting up mysql-server-5.7 (5.7.26-0ubuntu0.18.04.1) ...
Renaming removed key_buffer and myisam-recover options (if present)
ERROR: Unable to start MySQL server:
mysqld: Can't read dir of '/etc/mysql/conf.d/' (Errcode: 13 - Permission denied)
mysqld: [ERROR] Fatal error in defaults handling. Program aborted!
Please take a look at https://wiki.debian.org/Teams/MySQL/FAQ for tips on fixing common upgrade issues.
Once the problem is resolved, run apt-get --fix-broken install to retry.
dpkg: error processing package mysql-server-5.7 (--configure):
 installed mysql-server-5.7 package post-installation script subprocess returned error exit status 1
dpkg: dependency problems prevent configuration of mysql-server:
 mysql-server depends on mysql-server-5.7; however:
  Package mysql-server-5.7 is not configured yet.

dpkg: error processing package mysql-server (--configure):
 dependency problems - leaving unconfigured
No apport report written because the error message indicates its a followup error from a previous failure.
                          Errors were encountered while processing:
 mysql-server-5.7
 mysql-server
E: Sub-process /usr/bin/dpkg returned an error code (1)
ubuntu@ubuntu:~$ 

```

As you can see, I got an error after the command: ubuntu@ubuntu:~$ sudo apt-get install apt-transport-https

The error relates to mysql-server-5.7 not being configured. 

I am using sudo for the command but am getting *permission denied* error. I go to referenced folder in /etc and see this:

```
ubuntu@ubuntu:/etc/mysql$ ls
conf.d  debian-start  debian.cnf  my.cnf  my.cnf.fallback  mysql.cnf  mysql.conf.d
ubuntu@ubuntu:/etc/mysql$ ls conf.d
mysql.cnf  mysqldump.cnf
```

Thank you for any help. Please let me know what you would like me to try, and any additional information you might need.

**Comment by dagood** [comment]
Well written, thanks for all the info.

It looks like this was a problem with your system before running the steps--our packages don't depend on mysql, and a few things stand out:

> The following NEW packages will be installed:
>   apt-transport-https
> 0 upgraded, 1 newly installed, 0 to remove and 563 not upgraded.
> **2 not fully installed or removed.**

> Setting up mysql-server-5.7 (5.7.26-0ubuntu0.18.04.1) ...
> Renaming removed key_buffer and myisam-recover options (if present)
> ERROR: Unable to start MySQL server:
> mysqld: Can't read dir of '/etc/mysql/conf.d/' (Errcode: 13 - Permission denied)
> mysqld: [ERROR] Fatal error in defaults handling. Program aborted!
> Please take a look at https://wiki.debian.org/Teams/MySQL/FAQ for tips on fixing common upgrade issues.
> Once the problem is resolved, run apt-get --fix-broken install to retry.


Some searching turned up https://askubuntu.com/a/773443 as a way to try a purge/reinstall cycle, but I'm not familiar with that.

---

On the other hand, it doesn't look like `apt-transport-https` failed to install... have you tried continuing with the steps to get .NET Core installed?

**Comment by waqargithub** [comment]
Thanks for the quick and friendly response, and for the thoughtfulness to find and include the link about purge/reinstall cycle.

I did form the impression that mysql-server was part of the .NET installation so I did not try to continue the installation. I did proceed with the additional steps after seeing your post and was able to install .NET!

I have not been able to fully test it since there are other related missing installations to track down but they are not part of .NET. We can consider this issue closed.

Thanks again for your help!
